packages/governance/src/binaryBallotCounter.js

```diff
@@ -0,0 +1,187 @@
+  return Far('checker', { check });
+};
+
+// Exported for testing purposes
+const makeBinaryBallotCounter = (
+  question,
+  positions,
+  threshold,
+  tieOutcome = undefined,
+  closureRule,
+) => {
+  assert(
+    positions.length === 2,
+    X`Binary ballots must have exactly two positions. had ${positions.length}: ${positions}`,
+  );
+  assert.typeof(question, 'string');
+  assert.typeof(positions[0], 'string');
+  assert.typeof(positions[1], 'string');
+  if (tieOutcome) {
+    assert(
+      positions.includes(tieOutcome),
+      X`The default outcome on a tie must be one of the positions, not ${tieOutcome}`,
+    );
+  }
+
+  const template = buildBallot(ChoiceMethod.CHOOSE_N, question, positions, 1);
+  const ballotDetails = template.getDetails();
+
+  assert(
+    ballotDetails.method === ChoiceMethod.CHOOSE_N,
+    X`Binary ballot counter only works with CHOOSE_N`,
+  );
+  let isOpen = true;
+  const outcomePromise = makePromiseKit();
+  const tallyPromise = makePromiseKit();
+  const allBallots = makeStore('seat');
+
+  const recordBallot = (seat, filledBallot, shares = 1n) => {
+    assert(
+      filledBallot.question === question,
+      X`Ballot not for this question ${filledBallot.question} should have been ${question}`,
+    );
+    assert(
+      positions.includes(filledBallot.chosen[0]),
+      X`The ballot's choice is not a legal position: ${filledBallot.chosen[0]}.`,
+    );
+    allBallots.has(seat)
+      ? allBallots.set(seat, makeWeightedBallot(filledBallot, shares))
+      : allBallots.init(seat, makeWeightedBallot(filledBallot, shares));
+  };
+
+  const countVotes = () => {
+    assert(!isOpen, X`can't count votes while the election is open`);
+
+    // ballot template has position choices; Each ballot in allBallots should
+    // match. count the valid ballots and report results.
+    let spoiled = 0n;
+    const tally = {
+      [positions[0]]: 0n,
+      [positions[1]]: 0n,
+    };
+
+    allBallots.values().forEach(({ ballot, shares }) => {
+      assert(
+        ballot.chosen.length === 1,
+        X`A binary ballot must contain exactly one choice.`,
+      );
+      const choice = ballot.chosen[0];
+      if (!ballotDetails.positions.includes(choice)) {
+        spoiled += shares;
+      } else {
+        tally[choice] += shares;
+      }
+    });
+
+    const stats = {
+      spoiled,
+      votes: allBallots.entries().length,
+      results: [
+        { position: positions[0], total: tally[positions[0]] },
+        { position: positions[1], total: tally[positions[1]] },
+      ],
+    };
+    tallyPromise.resolve(stats);
+
+    if (!makeQuorumCounter(threshold).check(stats)) {
+      outcomePromise.reject('No quorum');
+      return;
+    }
+
+    if (tally[positions[0]] > tally[positions[1]]) {
+      outcomePromise.resolve(positions[0]);
+    } else if (tally[positions[1]] > tally[positions[0]]) {
+      outcomePromise.resolve(positions[1]);
+    } else {
+      outcomePromise.resolve(tieOutcome);
```

Comment:
In general there could also be "no quorum achieved" and "veto" in other voting arrangements.  I can easily imagine wanting standard answ4ers for those.

Reply:
Agreed, though I don't think it's necessary to support those in the first release.

Quorum counting is handled on line 107.

I think vetos would be a separate pass after the vote is complete. If there were a veto module included, that would indicate to voters that their vote isn't a final decision. Without it, the decision can be considered final. If the vote is automatically carried out, then it's truly final. Otherwise, the party that carries it out has the ability to delay or appeal to a higher consensus.